### DISCUSSION TITLE
Learning Fennel as the first Lisp

### DISCUSSION THREAD

**Discussion by molleweide** [ggandor/lightspeed.nvim#78] [Q&A] [answered]
Hi there,

I am excited to try out lightspeed. 

I have never used fennel before and I am wondering if you wrote the plugin in one big fennel file? 😃 
Personally I have just started out learning haskell which is pretty amazing and I am getting very curious
to learn some fennel as well. Is it hard to learn fennel and how much does knowledge of fennel transfer over to common lisp?
Would it be a good idea to look into fennel or should I learn lisp first?

**Comment by ggandor** [comment, marked as answer]
Hi! 🦉

> ...you wrote the plugin in one big fennel file?

Sure :)

> Personally I have just started out learning haskell which is pretty amazing

Indeed, keep on going! 😊

> Is it hard to learn fennel and how much does knowledge of fennel transfer over to common lisp? Would it be a good idea to look into fennel or should I learn lisp first?

Definitely dive into a more "proper" Lisp first - that can be Racket, Common Lisp, or Clojure (🤍), which latter is the most similar to Fennel _on the surface_. However, the emphasis is  on the "on the surface" part. To be able to use Fennel to its full potential, one has to be well-versed in (modern) Lisps _and_ Lua, since under the hood Fennel is all Lua - it's mostly a syntactic convenience layer. That said, given those two prerequisites it's a breeze to learn it, with only one or two minor gotchas ahead.

**Reply by molleweide** [reply]
dude thanks for replying and sorry for my late response. i might have to checkout out how to make some basic clojure cli then to learn the basics. this is going to be fun!

**Reply by MuhammedZakir** [reply]
If you're going for Common Lisp, I recommend https://gigamonkeys.com/book/. It's really good. :-)

**Reply by molleweide** [reply]
woaw wait it's you muhammed!! 😄  thanks for the link.

**Reply by ggandor** [reply]
Agreed, Practical Common Lisp is a fantastic book.

**Reply by molleweide** [reply]
hey guy do you mostly use emacs or vim?

**Reply by ggandor** [reply]
If the question was addressed to me, then mostly Neovim :)

**Reply by molleweide** [reply]
oh I misspelled guys 😅

**Reply by MuhammedZakir** [reply]
I use Neovim too. I also sometimes use Emacs, mainly when my terminal emulator doesn't support or handles font poorly. E.g. non-monospace fonts. ~But even for that, I usually open Neovim inside Emacs. :-P Because of issues like this, I'm thinking about using Emacs as my terminal emulator, but hasn't found a good config and don't have the time/mood for writing it myself.~

**Reply by molleweide** [reply]
You can run neovim inside of emacs wtf?

Can you run emacs in emacs lol?

**Reply by MuhammedZakir** [reply]
Don't do that. Take it as a joke. It won't work like you would expect.

About Emacs: I use Evil layer. I heard Doom config is pretty good if you want Vim-like experience in Emacs and it's also a batteries-included config -- comes with most of the stuff that you will need.

**Reply by molleweide** [reply]
Haha cool yeah i get it but it sounds funny. I use doom-nvim instead so that I can get help with configs and stuff since I am quite new to vim.

**Reply by MuhammedZakir** [reply]
I didn't know about doom-nvim. Thanks for that! :-) FYI, there is another nice config named LunarVim.

**Reply by molleweide** [reply]
Yeah i saw lunarnvim also but had already gotten invested in doom nvim. In the end theyre probably very similar but i really like doom nvim

**Reply by molleweide** [reply]
The maintainers of doom are also the guys developing neorg which is suppose to be like an org equivalent for nvim
